### PR TITLE
Add run immediately option databricks

### PR DESCRIPTION
**Pull request by danielvdende**
## Summary:

As mentioned in #55 there are cases in which you submit a streaming job and do not want to run it immediately. This PR adds functionality to `deploy_to_databricks` to support this. It adds an optional parameter to the `deploy_to_databricks` step, which gives the user control of whether or not to run the streaming job immediately. The default value is `True`.

It also bumps the version of `gitpython` as a minor side-effect, as the previous versions had issues (documented here: https://github.com/gitpython-developers/GitPython/issues/983) 

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] There is no commented out code in this PR.
  - [x] There are no new TODOs in this PR

If exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated

### CODE REVIEW

**Comment by danielvdende** [comment]
@Jorricks could you take a look at this? I think this resolves #55, would be great if you could confirm.

File: tests/azure/test_deploy_to_databricks.py [hunk 1]
```diff
@@ -599,12 +599,14 @@ def test_kill_it_with_fire(self, victim):
...
-        victim._submit_job({}, False)
+        victim._submit_job({}, False, True)
 
         victim.jobs_api.create_job.assert_called_with({})
 
+        victim.jobs_api.run_now.assert_not_called()
+
     def test_submit_job_streaming(self, victim):
-        victim._submit_job({}, True)
+        victim._submit_job({}, True, True)
```
Comment (timvancann) [inline tests/azure/test_deploy_to_databricks.py:609]: Semantically "submit_job" branches  a bit too often I feel. The wording indicates a job is submitted, period. Perhaps it makes sense to split this "submit_job" to `create_job` and `run_job`. This also keeps it in line with DataBricks teminology.

Comment (danielvdende) [reply tests/azure/test_deploy_to_databricks.py]: Good point. I refactored a bit. What do you think now?

Comment (timvancann) [reply tests/azure/test_deploy_to_databricks.py]: Looks good 👍

**Pull request merged**